runner/src/main.rs

```diff
@@ -125,125 +124,153 @@ fn is_cargo_workspace_file(path: &PathBuf) -> bool {
-        &["build", "--", "//oak/...:all", "-//oak/server/asylo:all"],
-    )
-}
-
-fn run_bazel_test() -> R {
-    step(
-        "host targets",
-        "bazel",
-        // TODO: Extract these targets with `bazel query` at runtime, based on some label or
-        // attribute.
-        &[
-            "test",
-            "//oak/server:host_tests",
-            "//oak/server/storage:host_tests",
-            "//oak/common:host_tests",
-        ],
-    )
+            .map(|entry| Step::Single {
+                name: entry.clone(),
+                runnable: cmd("embedmd", &["-d", &entry]),
+            })
+            .collect(),
+    }
+}
+
+fn run_cargo_fmt() -> Step {
+    Step::Multiple {
+        name: "cargo fmt".to_string(),
+        steps: workspace_manifest_files()
+            .map(to_string)
+            .map(|entry| Step::Single {
+                name: entry.clone(),
+                runnable: cmd(
+                    "cargo",
+                    &[
+                        "fmt",
+                        "--all",
+                        &format!("--manifest-path={}", &entry),
+                        "--",
+                        "--check",
+                    ],
+                ),
+            })
+            .collect(),
+    }
+}
+
+fn run_cargo_test() -> Step {
+    Step::Multiple {
+        name: "cargo test".to_string(),
+        steps: workspace_manifest_files()
+            .map(to_string)
+            .map(|entry| Step::Single {
+                name: entry.clone(),
+                runnable: cmd(
+                    "cargo",
+                    &[
+                        "test",
+                        "--all-targets",
+                        &format!("--manifest-path={}", &entry),
+                    ],
+                ),
+            })
+            .collect(),
+    }
+}
+
+fn run_cargo_doc_test() -> Step {
+    Step::Multiple {
+        name: "cargo doc test".to_string(),
+        steps: workspace_manifest_files()
+            .map(to_string)
+            .map(|entry| Step::Single {
+                name: entry.clone(),
+                runnable: cmd(
+                    "cargo",
+                    &["test", "--doc", &format!("--manifest-path={}", &entry)],
+                ),
+            })
+            .collect(),
+    }
+}
+
+fn run_cargo_clippy() -> Step {
+    Step::Multiple {
+        name: "cargo clippy".to_string(),
+        steps: workspace_manifest_files()
+            .map(to_string)
+            .map(|entry| Step::Single {
+                name: entry.clone(),
+                runnable: cmd(
+                    "cargo",
+                    &[
+                        "clippy",
+                        "--all-targets",
+                        &format!("--manifest-path={}", &entry),
+                        "--",
+                        "--deny=warnings",
+                    ],
+                ),
+            })
+            .collect(),
+    }
+}
+
+fn run_bazel_build() -> Step {
```

Comment:
Should it be called `run_bazel_build_all`?

Reply:
I'll rename later when I add more precise rules anyways.